src/main/dialogs/captureToolDialog.js

```diff
@@ -0,0 +1,294 @@
+
+    var startupTime = Date.now();
+    var captureDirName = gpiiSettingsDir + "/capture-diagnostics-" + gpii.journal.formatTimestamp(startupTime);
+
+    fluid.log("Capture Diagnostics directory is: ", captureDirName);
+
+    var dailogPromise = electron.dialog.showMessageBox({
+        buttons: ["Cancel", "Run Capture"],
+        message: "Select 'Run Capture' to run the capture diagnostics and store them to disk.\n\n  After capture they will be stored in:\n" + captureDirName
+    });
+
+    if (dailogPromise === 1) {
+        fs.mkdirSync(captureDirName);
+        gpii.app.captureTool.generateDiagnostics(captureToolUtils).then(
+            function (data) {
+                // 0. Solutions
+                fs.appendFileSync(captureDirName + "/solutions.json", JSON.stringify(data.solutions, null, 4));
+                // 1. Installed Solutions
+                fs.appendFileSync(captureDirName + "/installedSolutions.json", JSON.stringify(data.installedSolutions, null, 4));
+                // 2. Full System Settings Capture
+                fs.appendFileSync(captureDirName + "/settingsCapture.json", JSON.stringify(data.settingsCapture, null, 4));
+
+                electron.dialog.showMessageBox({
+                    message: "Capture diagnostics successfully taken and are now stored in this directory:\n" + captureDirName
+                });
+            },
+            function (err) {
+                electron.dialog.showMessageBox({
+                    message: "There was a problem capturing the diagnostics:\n" + err
+                });
+                fluid.log("Error running capture diagnostics. ", err);
+            }
+        );
+    }
+};
+
+/*
+ * Generates and returns a set of diagnostics for the current local machine.
+ *
+ * Returns a promise containing these.
+ */
+gpii.app.captureTool.generateDiagnostics = function (captureToolUtils) {
+    return fluid.promise.fireTransformEvent(captureToolUtils.events.onCollectDiagnostics, {});
+};
+
+gpii.app.captureTool.init = function (that, flowManager) {
+    ipcMain.on("getInstalledSolutions", function (event /*, arg */) {
+        flowManager.capture.getInstalledSolutions().then(
+            function (data) {
+                event.sender.send("sendingInstalledSolutions", data);
+                var runningSolutions = {};
+                fluid.each(data, function (solutionEntry, solutionId) {
+                    try {
+                        fluid.log("ABOUT TO CHECK PROCESS FOR: ", solutionId);
+                        var isRunningState = gpii.processReporter.handleIsRunning(solutionEntry);
+                        fluid.log("RUNNING STATE: ", isRunningState);
+                        if (isRunningState) {
+                            runningSolutions[solutionId] = solutionEntry;
+                        }
+                    } catch (err) {
+                        fluid.log("Exception trying to look up", solutionEntry.isRunning);
```

Comment:
Unlikely that this message will be meaningful. At least log the exception somehow, and also check whether "isRunning" is even an existing string member

Reply:
added more detail to this message